File: scripts/verify_auth.js
const http = require('http');

function request(path, method = 'GET', body, cookie) {
  return new Promise((resolve, reject) => {
    const payload = body ? JSON.stringify(body) : null;
    const headers = {
      'Content-Type': 'application/json',
      ...(cookie ? { Cookie: cookie } : {})
    };
    if (payload) headers['Content-Length'] = Buffer.byteLength(payload);

    const req = http.request({ host: 'localhost', port: 8180, path, method, headers }, (res) => {
      let data = '';
      res.setEncoding('utf8');
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve({ status: res.statusCode, headers: res.headers, body: data }));
    });
    req.on('error', reject);
    if (payload) req.write(payload);
    req.end();
  });
}

(async () => {
  const email = `verify-flow-${Date.now()}@example.com`;
  const registerRes = await request('/api/session/register', 'POST', {
    email,
    password: 'secret123',
    firstName: 'Verify',
    lastName: 'Flow',
    phone: '123456789'
  });

  console.log('register status:', registerRes.status);
  console.log('register body:', registerRes.body);
  const setCookie = (registerRes.headers['set-cookie'] || [])[0] || '';
  console.log('set-cookie:', setCookie);

  const meRes = await request('/api/session/me', 'GET', null, setCookie.split(';')[0]);
  console.log('me status:', meRes.status);
  console.log('me body:', meRes.body);
})();

File: scripts/verify_auth.py
import json
import http.client

conn = http.client.HTTPConnection('localhost', 8180, timeout=10)
payload = json.dumps({
    'email': 'verify-flow-python@example.com',
    'password': 'secret123',
    'firstName': 'Verify',
    'lastName': 'Python',
    'phone': '123456789'
})
headers = {'Content-Type': 'application/json'}
conn.request('POST', '/api/session/register', body=payload, headers=headers)
res = conn.getresponse()
body = res.read().decode('utf-8')
print('register_status', res.status)
print('register_body', body)
set_cookie = res.getheader('Set-Cookie') or ''
print('set_cookie', set_cookie)

conn2 = http.client.HTTPConnection('localhost', 8180, timeout=10)
headers2 = {}
if set_cookie:
    headers2['Cookie'] = set_cookie.split(';')[0]
conn2.request('GET', '/api/session/me', headers=headers2)
res2 = conn2.getresponse()
body2 = res2.read().decode('utf-8')
print('me_status', res2.status)
print('me_body', body2)
